packages/model/src/aspects/Constraint.ts

```diff
@@ -89,12 +89,41 @@ export class Constraint extends Aspect<Constraint.Definition> implements Constra
      */
     test(value: FieldValue, properties?: Record<string, any>): boolean {
         // Helper that looks up "reference" field values in properties.  This is for constraints such as "min FieldName"
-        function valueOf(value: unknown, raw = false) {
+        function valueOf(value: unknown, raw = false): unknown {
             if (!raw && (typeof value === "string" || Array.isArray(value))) {
                 return value.length;
             }
             if (isObject(value)) {
                 const { type, name } = value;
+                switch (type) {
+                    case FieldValue.Reference:
+                        if (typeof name === "string") {
+                            value = valueOf(properties?.[camelize(name)], raw);
+                        }
+                        break;
+
+                    case "+":
+                        {
+                            const lhs = valueOf(value.lhs);
+                            const rhs = valueOf(value.rhs);
+                            if (typeof lhs === "number" && typeof rhs === "number") {
+                                return lhs + rhs;
+                            }
+                            return undefined;
+                        }
+                        break;
+
+                    case "-":
+                        {
+                            const lhs = valueOf(value.lhs);
+                            const rhs = valueOf(value.rhs);
+                            if (typeof lhs === "number" && typeof rhs === "number") {
+                                return lhs - rhs;
+                            }
+                            return undefined;
+                        }
+                        break;
+                }
                 if (type === FieldValue.reference && typeof name === "string") {
```

Comment:
Is this supposed to be removed because now duplicated?

Reply:
Well yes, it should have been removed because it *should* have been a duplicate, but there was a bug in the version above.  Fixed now